### ISSUE TITLE
add default warnings

### ISSUE THREAD

**Issue by decioferreira** [guida-lang/compiler#45]
introduce warnings to the compiler. Currently we have the following:

```
type Warning
    = UnusedImport A.Region Name
    | UnusedVariable A.Region Context Name
    | MissingTypeAnnotation A.Region Name Can.Type
```

but they are not being reported.
